wallet/src/account/output_cache/mod.rs

```diff
@@ -68,6 +68,22 @@ impl OutputCache {
         &self.consumed
     }
 
+    pub fn pool_ids(&self) -> BTreeSet<PoolId> {
+        self.txs
```

Comment:
Perhaps this can be optimized later by making a specific key/value store for staking pools. A wallet growing to thousands of transactions can easily make this process unnecessarily complex and hard.

Reply:
added a map for the pool ids to creating block heigh and timestamp